Src/RE2.Managed/Regex2.cs

```diff
@@ -191,108 +191,86 @@ public static void ClearRegexes()
-                out int numNamedCapturingGroups,
-                out int groupNamesBufferSize);
-            int numSubmatches = numCapturingGroups + 1;
-
             byte[] patternUtf8Bytes = Encoding.UTF8.GetBytes(pattern);
             byte[] textUtf8Bytes = Encoding.UTF8.GetBytes(text);
-            GroupNameHeader[] groupNameHeaders = new GroupNameHeader[numNamedCapturingGroups];
-            byte[] groupNamesBuffer = new byte[groupNamesBufferSize];
-            Submatch[] submatches = new Submatch[numSubmatches];
 
             fixed (byte* patternUtf8BytesPtr = patternUtf8Bytes)
             fixed (byte* textUtf8BytesPtr = textUtf8Bytes)
-            fixed (GroupNameHeader* groupNameHeadersPtr = groupNameHeaders)
-            fixed (byte* groupNamesBufferPtr = groupNamesBuffer)
-            fixed (Submatch* submatchesPtr = submatches)
             {
-                bool isMatch =
-                    NativeMethods.MatchesNamedGroups(
-                        new StringUtf8(patternUtf8BytesPtr, pattern.Length),
-                        new StringUtf8(textUtf8BytesPtr, text.Length),
-                        groupNameHeadersPtr,
-                        groupNamesBufferPtr,
-                        submatchesPtr);
-                if (isMatch)
-                {
-                    groupName2Index = new Dictionary<string, int>();
-                    index2GroupName = new Dictionary<int, string>();
+                MatchesCaptureGroupsOutput* output;
 
-                    // Build GroupName-Index maps
-                    int groupNameStringIndex = 0;
-                    foreach (GroupNameHeader groupNameHeader in groupNameHeaders)
-                    {
-                        string groupName = Encoding.UTF8.GetString(groupNamesBuffer, groupNameStringIndex, groupNameHeader.Length);
-                        groupName2Index[groupName] = groupNameHeader.Index;
-                        index2GroupName[groupNameHeader.Index] = groupName;
-                        groupNameStringIndex += groupNameHeader.Length;
-                    }
+                NativeMethods.MatchesCaptureGroups(
+                    new StringUtf8(patternUtf8BytesPtr, pattern.Length),
+                    new StringUtf8(textUtf8BytesPtr, text.Length),
+                    &output);
 
-                    // Build submatch list
-                    submatchStrings = new List<string>();
-                    foreach (Submatch submatch in submatches)
-                    {
-                        Console.WriteLine(submatch.Index);
-                        string submatchString = Encoding.UTF8.GetString(textUtf8Bytes, submatch.Index, submatch.Length);
-                        submatchStrings.Add(submatchString);
-                    }
-
-                    return true;
-                }
-                else
+                // Build SubmatchIndex to GroupName map
+                var submatchIndex2GroupName = new Dictionary<int, string>();
```

Comment:
>var submatchIndex2GroupName = new Dictionary<int, string>(); [](http://example.com/codeflow?start=16&length=60)

you could add the size to the constructor, to prevent resizing

Reply:
Done.